WordPress/gutenberg
# Allow styles to be changed dynamically through editor settings

## What?


Fixes https://github.com/WordPress/gutenberg/pull/50091#issuecomment-1592582675.

## Why?


It should be possible to add content styles dynamically in JS. This should be possible through `wp.data.dispatch( 'core/editor' ).updateEditorSettings`, but it's currently broken: the styles passed through the `Layout` component (and all the way down to `Iframe` are the initialised styles, not the styles from the store, so these components don't react to updates.

This PR fixes that and adds an e2e test.

## How?


## Testing Instructions





There's an e2e test, but you can try using `wp.data.dispatch( 'core/editor' ).updateEditorSettings` (see e2e test).

### Testing Instructions for Keyboard


## Screenshots or screencast
